src/mbgl/renderer/raster_bucket.cpp

```diff
@@ -12,8 +13,7 @@ RasterBucket::RasterBucket(PremultipliedImage&& image_) : image(std::move(image_
 }
 
 void RasterBucket::upload(gl::Context& context) {
-    texture = context.createTexture(image);
-    image = {};
+    texture = context.createTexture(util::unpremultiply(std::move(image)));
```

Comment:
Yeah, it's lossy ad might result it tiny differences in some cases with non-default style properties. They probably wouldn't be human-noticeable.

It would be better to not do this, but your call.